Sources/Attribution/AttributionPoster.swift

```diff
@@ -106,8 +109,28 @@ class AttributionPoster {
             return
         }
 
-        Logger.debug("Logging attribution token for now to avoid lint warning: \(attributionToken)")
-        // post
+        self.post(adServicesToken: attributionToken)
+    }
+
+    func post(adServicesToken: String) {
+        let currentAppUserID = self.currentUserProvider.currentAppUserID
+
+        // set the cache in advance to avoid multiple post calls
+        let latestNetworkIdsAndAdvertisingIdsSentByNetwork =
+            self.deviceCache.latestNetworkAndAdvertisingIdsSent(appUserID: currentAppUserID)
+        var newDictToCache = latestNetworkIdsAndAdvertisingIdsSentByNetwork
+        newDictToCache[String(AttributionNetwork.adServices.rawValue)] = adServicesToken
+        self.deviceCache.set(latestNetworkAndAdvertisingIdsSent: newDictToCache, appUserID: currentAppUserID)
+
+         backend.post(adServicesToken: adServicesToken, appUserID: currentAppUserID) { error in
+             if let receivedNSError = error as NSError? {
+                 Logger.warn(Strings.attribution.adservices_token_post_failed(error: receivedNSError))
```

Comment:
should we just have this accept an `Error` so we don't need the cast? 
Also, is `receivedNSError` a `NSError?` at this point? if so, this would log even if `receivedNSError` is `nil`, right?

Reply:
it's a `BackendError`. i remember thinking there was a reason for the cast but after PTO i'm not remembering what that was.. switching to BackendError